thunder/torch/__init__.py

```diff
@@ -4710,18 +4710,21 @@ def embedding(
 ) -> TensorLike:
     # TODO: add embedding_renorm_ so we can remove embedding prim
     # NOTE: padding_idx has impact on backward and is not supported by take
-    if max_norm is not None or padding_idx is not None:
-        padding_idx = padding_idx if padding_idx is not None else -1
-        return prims.embedding(
-            a,
-            weight,
-            padding_idx=padding_idx,
-            max_norm=max_norm,
-            norm_type=norm_type,
-            scale_grad_by_freq=scale_grad_by_freq,
-            sparse=sparse,
+    if max_norm is not None:
+        # See https://github.com/Lightning-AI/lightning-thunder/issues/1699
+        raise NotImplementedError(
+            "max_norm argument is currently not support. Please create an issue detailing your use case."
         )
 
+    if padding_idx is not None:
+        padding_idx = padding_idx if padding_idx is not None else -1
+
+        utils.check(a.dtype == dtypes.int64, lambda: f"Expected a.dtype={a.dtype} to be int64")
+        utils.check(weight.ndim == 2, lambda: f"Expected weight (weight.shape={weight.shape} to be a matrix)")
+        shape = list(a.shape)
+        shape.append(weight.shape[1])
+        return TensorProxy(like=weight, shape=shape)
```

Comment:
Maybe want to add a comment for the direct construction of a tensorproxy here instead of calling an operator that would create one -- I'm curious, too

Reply:
`thunder.torch. embedding` is the operator we're calling to create the proxy.